cmake/macros/macro_populate_target_properties.cmake

```diff
@@ -87,6 +87,18 @@ function(populate_target_properties _target _build)
     ${DEAL_II_DEFINITIONS} ${DEAL_II_DEFINITIONS_${_build}}
     )
 
+  #
+  # Add target properties:
+  #
+  #  - set POSITION_INDEPENDENT_CODE to true to compile everything with
+  #    the -fpic/-fPIC compiler flag. This ensures that we can link all
+  #    object targets into a relocatable library at the end.
+  #
+
+  set_target_properties(${_target} PROPERTIES
+    POSITION_INDEPENDENT_CODE TRUE
+    )
```

Comment:
But this is only necessary for shared libraries, right? Is this in a block that only pertains to shared libs?

Reply:
But you *don't* need to set that property for static archives. (It isn't *wrong* for static archives, but the code is -- marginally -- slower.)

Reply:
@bangerth Correct, at least until someone tries to link the static archive into a shared object again :smile:

Reply:
@tjhei For the modules? Yup. (It is redundant because CMake defaults to enabling `POSITION_INDEPENDENT_CODE` for `SHARED` objects but it doesn't hurt to set it automatically.)

Reply:
Oh. We only support shared libs as ASPECT plugins. So I guess we can close the PR.